src/mlpack/methods/adaboost/adaboost_impl.hpp

```diff
@@ -246,7 +245,6 @@ void AdaBoost<WeakLearnerType, MatType>::serialize(Archive& ar,
 {
   ar & BOOST_SERIALIZATION_NVP(numClasses);
   ar & BOOST_SERIALIZATION_NVP(tolerance);
-  ar & BOOST_SERIALIZATION_NVP(ztProduct);
```

Comment:
Ah, there is a little bit of trickiness here... we need to preserve reverse compatibility: if a user has an AdaBoost model saved with an old version, and we try to load with this version, it will crash.  Here's some code to do that:

```
template<typename WeakLearnerType, typename MatType>
template<typename Archive>
void AdaBoost<WeakLearnerType, MatType>::serialize(Archive& ar,
const unsigned int version)
{
  ar & BOOST_SERIALIZATION_NVP(numClasses);
  ar & BOOST_SERIALIZATION_NVP(tolerance);
  if (version == 0 && Archive::is_loading::value)
  {
    // Load unused ztProduct double and forget it.
    double tmpZtProduct = 0.0;
    ar & BOOST_SERIALIZATION_NVP(ztProduct);
  }
  ar & BOOST_SERIALIZATION_NVP(alpha);
}
```

Then, outside the class, you can add this (adapted from `src/mlpack/methods/ffn/ffn.hpp`):

```
//! Set the serialization version of the FFN class.  Multiple template arguments
//! makes this ugly...
namespace boost {
namespace serialization {

template<typename WeakLearnerType,
         typename MatType>
struct version<
    mlpack::adaboost::AdaBoost<WeakLearnerType, MatType>>
{
  BOOST_STATIC_CONSTANT(int, value = 1);
};

} // namespace serialization
} // namespace boost
```

Reply:
Okay. 
With this I also got a chance to know boost serialization. Thanks :+1: